src/topics/lda_model.cpp

```diff
@@ -16,30 +17,23 @@ lda_model::lda_model(const learn::dataset& docs, std::size_t num_topics)
     /* nothing */
 }
 
-void lda_model::save_doc_topic_distributions(const std::string& filename) const
+void lda_model::save_doc_topic_distributions(std::ostream& stream) const
 {
-    std::ofstream file{filename};
-    for (const auto& doc : docs_)
+    io::packed::write(stream, docs_.size());
+    io::packed::write(stream, num_topics_);
+
+    for (const auto& d : docs_)
     {
-        file << doc.id << "\t";
-        double sum = 0;
+        auto dist = topic_distrbution(doc_id{d.id});
         for (topic_id j{0}; j < num_topics_; ++j)
         {
-            double prob = compute_doc_topic_probability(doc.id, j);
-            if (prob > 0)
-                file << j << ":" << prob << "\t";
-            sum += prob;
+            io::packed::write(stream, dist.counts(j));
```

Comment:
Nope. I was hoping to use the probabilities, but in order to reconstruct the distribution you can only do so with the counts so far as I can tell.

That said, I didn't notice that multinomial implemented those functions which is great and I've moved to that.